[LinWanLeii] iP

LinWanLeii opened the pull request:
Feeling lonely and in need for someone to manage your Tasks.
**CathyTheChattyCat** is the one for you.

- Simple Texting formats
- ~~NO NEED~~ Easy to learn
- Run and use

All you need to do is,

1. Download it from [here](https://github.com/LinWanLeii/ip.git).(No virus no problem)
2. Run and add a text file you wanna store it at.
3. And store as many task as you want 

MOST IMPORTANTLY,
`IT IS FREE 💰`

Features:

- [ ] Manges Task
- [ ] Store Tasks
- [ ] Error Detecting

If you Java programmer, you can use it to practice Java too. Here's the main method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

LinWanLeii committed:
Done till lvl 4

LinWanLeii committed:
Level-5

LinWanLeii committed:
Level-6

LinWanLeii committed:
Level-0

LinWanLeii committed:
Level-1

LinWanLeii committed:
Level-2

LinWanLeii committed:
Level-6

LinWanLeii committed:
Update body

LinWanLeii committed:
Add writing and reading file function

LinWanLeii committed:
Add Local Dates and times

LinWanLeii committed:
Add More OOP

LinWanLeii committed:
Add Packages

LinWanLeii committed:
Merge branch 'add-gradle-support' of https://github.com/LinWanLeii/ip

LinWanLeii committed:
Add testcases and Jar

LinWanLeii committed:
Add Javadocs

LinWanLeii committed:
Fix JavaDocs

LinWanLeii committed:
Add find function

LinWanLeii committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add find function

# Conflicts:
#	Testing.txt

dloh2236 commented on src/main/java/Parser/Parser.java:
```diff
@@ -0,0 +1,111 @@
+    public static Command parse(String fullCommand) throws DukeException{
+        String[] split = fullCommand.split(" ", 2);
+        String commandAction = split[0];
+        switch (commandAction) {
+            case "list":
```
switch case should be inline with no indents for case clauses as per java coding standard

dloh2236 commented on src/main/java/Parser/Parser.java:
```diff
@@ -0,0 +1,111 @@
+                }
+            case "todo":
+                if (split.length < 2 || split[1].isEmpty()) {
+                    throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
```
can just use your declared and initialised UNKNOWN_COMMAND

dloh2236 commented on src/main/java/Tasks/Task.java:
```diff
@@ -0,0 +1,57 @@
+    public String type;
+    public Task(String name) {
+        this.name = name;
+        this.marked = false;
```
boolean var should sound like a boolean

dloh2236 reviewed (commented):
small changes :D

LinWanLeii committed:
add checkstyle

saraozn commented on src/main/java/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,41 @@
+ */
+public class DeadlineCommand extends Command {
+    private final String task;
+    private final String deadDate;
```
Use correct spelling in names, `deadline` works!

saraozn commented on src/main/java/command/ByeCommand.java:
```diff
@@ -0,0 +1,26 @@
+     * @param ui contains the user interface that will be shown to the user depending on the inputs.
+     * @param fileStorage Writing and reading on text files.
+     */
+    public void excute(TaskList taskList, Ui ui, FileStorage fileStorage) {
```
Use correct spelling in names

saraozn commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,154 @@
+    private Scanner scanner;
+    final String UNKNOWN_COMMAND = "☹ OOPS!!! I'm sorry, but I don't know what that means :-(";
+    final String NAME = "CathyTheChattyCat";
+    String lineBreak = "\n__________________________________________\n";
```
```suggestion
    final String LINE_BREAK = "\n__________________________________________\n";
```

saraozn commented on src/test/java/seedu/duke/TaskTest.java:
```diff
@@ -0,0 +1,28 @@
+
+public class TaskTest {
+    @Test
+    public void simpleToDoTest(){
```
```suggestion
    public void simpleToDoTest() {
```

saraozn commented on src/main/java/list/TaskList.java:
```diff
@@ -0,0 +1,124 @@
+     *
+     * @return the size of the TaskList.
+     */
+    public int size() {
```
use verbs for methods

saraozn reviewed (commented):
Few minor changes that can be made, otherwise looks good!

LinWanLeii committed:
Added Gui for CathyTheChattyCat

LinWanLeii committed:
Update build.gradle file
Need to add a Jar file through shadowjar

This is to ensure that the application will run even if the user does not have JavaFX on their device

ShadowJar from gradle was being used

Let's ensure that we use gradle to its best ability because it is good

LinWanLeii committed:
Added Assertions and edited FileStorage file

LinWanLeii committed:
Merge pull request #1 from LinWanLeii/Branch-A-Assertions

Added Assertions and edited FileStorage file

LinWanLeii committed:
Updated CodeQuality, Assertions and Javadocs
Make some edits on codes to improve the quality and readability, update more useful assertions and better javadocs descriptions.

This is so that anyone can easily understand what i happening to my code when reading.

Let's edit the code and ensure that everything is up to standard.

Standards have to be upholded

LinWanLeii committed:
Merge pull request #2 from LinWanLeii/Branch-A-CodeQuality

Updated CodeQuality, Assertions and Javadocs

LinWanLeii committed:
Added C-MassOps but only for delete command and Added Streams

LinWanLeii committed:
Added more error handling and added multiple command for mark and unmark

LinWanLeii committed:
Merge pull request #4 from LinWanLeii/Branch-A-MoreErrorHandling

Added more error handling and added multiple command for mark and unmark

LinWanLeii committed:
Added UserGuide

LinWanLeii committed:
Updated gradle

LinWanLeii committed:
Updated some check style and edited multiple command file to reduce code length

LinWanLeii committed:
Updated some errors mentioned by the bot

LinWanLeii committed:
Added more details in readme

LinWanLeii committed:
Solved bug relating to checkstyle and gradle
